### PR TITLE
Fix earlystopping metric issue #3722

### LINKED ISSUE #3722: EarlyStoppingListener don't work.
public static DefaultTrainingConfig buildTrainingConfig() {
        return new DefaultTrainingConfig(Loss.softmaxCrossEntropyLoss())
                .optOptimizer(Adam.builder()
                        .optClipGrad(0.5f)
                        .optWeightDecays(5e-3f)
                        .optLearningRateTracker(
                                Tracker.multiFactor()
                                        .setBaseValue(5e-3f)
                                        .setSteps(new int[]{10, 15, 20})
                                        .optFactor(0.5f)
                                        .build())
                        .build())
                .addEvaluator(new Accuracy())
                .addTrainingListeners(TrainingListener.Defaults.logging())
                .addTrainingListeners(EarlyStoppingListener.builder()
                        .optEpochPatience(1)
                        .optEarlyStopPctImprovement(1)
                        .optMaxDuration(Duration.ofMinutes(42))
                        .optMinEpochs(5)
                        .build()
                );
    }
Epoch 00 | Train Loss: 1.0994 | Train Acc: 0.3500 | Val Acc: 0.3716
Epoch 01 | Train Loss: 1.0984 | Train Acc: 0.3417 | Val Acc: 0.3716
Epoch 02 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 03 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 04 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 05 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 06 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 07 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 08 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 09 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 10 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 11 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 12 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 13 | Train Loss: 1.0982 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 14 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 15 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 16 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 17 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 18 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 19 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 20 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 21 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 22 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 23 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 24 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 25 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 26 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 27 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 28 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 29 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 30 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 31 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 32 | Train Loss: 1.0982 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 33 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 34 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 35 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 36 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 37 | Train Loss: 1.0985 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 38 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 39 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 40 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 41 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 42 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 43 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 44 | Train Loss: 1.0982 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 45 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 46 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 47 | Train Loss: 1.0983 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 48 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716
Epoch 49 | Train Loss: 1.0984 | Train Acc: 0.3437 | Val Acc: 0.3716

### PR DESCRIPTION
**Pull request by SamBSalgado**
## Description
This PR fixes issue #3722  by extending the `EarlyStoppingListener` to support `"trainLoss"` as a valid monitored metric in addition to the default `"validateLoss"`.

Previously, if a user specified `"trainLoss"` via `.optMonitoredMetric("trainLoss")`, the early stopping logic would not behave as expected because `TrainingResult.getTrainLoss()` was not considered in the metric selection. This update introduces an explicit check for `"trainLoss"` in the `getMetric(...)` method, ensuring it is properly used for early stopping decisions.

### Changes Summary
- The `getMetric(...)` method now explicitly supports `"trainLoss"` in addition to `"validateLoss"`.

### Result
- When using `.optMonitoredMetric("trainLoss")`, training is now stopped early if the training loss fails to improve as specified.
- This improves flexibility for users who want to monitor training loss instead of validation loss.

Closes #3722

### CODE REVIEW

**Comment by frankfliu** [comment]
Please fix java format issue:

```
./gradlew fJ
```

**Review by frankfliu — APPROVED** [review]
